cashu/mint/crud.py

```diff
@@ -264,3 +268,28 @@ async def get_keyset(
         tuple(values),
     )
     return [MintKeyset(**row) for row in rows]
+
+
+async def store_invoice_and_proofs(
+    db: Database,
+    invoice: bolt11_invoice,
+    preimage: str,
+    proof: Proof,
+    conn: Optional[Connection] = None,
+):
+    await (conn or db).execute(
+        f"""
+            INSERT INTO {table_with_schema(db, 'payments')}
+              (invoice_amount, date, payment_hash, preimage, proof_amount, proof_C, proof_secret)
+            VALUES (?, ?, ?, ?, ?, ?, ?)
+            """,
+        (
+            invoice.amount_msat / 1000,
```

Comment:
Perhaps worth the explicit 'cast' to an int to indicate that's what's going to happen DB-side anyway?
```suggestion
            int(invoice.amount_msat / 1000),
```